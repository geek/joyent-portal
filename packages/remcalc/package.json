{
  "name": "remcalc",
  "version": "1.0.5",
  "license": "MPL-2.0",
  "description": "Calculate the `rem`'s from `px` values",
  "keywords": [
    "calc",
    "rem",
    "em",
    "px",
    "pixels",
    "pixel"
  ],
  "repository": "github:yldio/joyent-portal",
  "main": "dist/remcalc.umd.js",
  "jsnext:main": "dist/remcalc.es.js",
  "module": "dist/remcalc.es.js",
  "entry": "src/index.js",
  "scripts": {
    "lint": "eslint . --fix",
    "lint-ci": "eslint . --format junit --output-file $CIRCLE_TEST_REPORTS/lint/remcalc.xml",
    "test": "bup && cross-env NODE_ENV=test nyc --reporter=lcov --reporter=text ava",
    "test-ci": "cross-env NODE_ENV=test nyc --report-dir=$CIRCLE_ARTIFACTS/remcalc --reporter=lcov --reporter=text ava --tap | tap-xunit > $CIRCLE_TEST_REPORTS/test/remcalc.xml",
    "build": "bup",
    "prepublish": "bup"
  },
  "dependencies": {
    "lodash.flatten": "^4.4.0"
  },
  "devDependencies": {
    "ava": "0.19.1",
    "babel-plugin-istanbul": "^4.1.4",
    "babel-plugin-transform-es2015-arrow-functions": "^6.22.0",
    "babel-plugin-transform-es2015-parameters": "^6.24.1",
    "babel-plugin-transform-es2015-spread": "^6.22.0",
    "babel-plugin-transform-es2015-template-literals": "^6.22.0",
    "babel-register": "^6.24.1",
    "bup": "^1.0.9",
    "cross-env": "^5.0.1",
    "eslint": "^3.19.0",
    "eslint-config-joyent-portal": "1.0.0",
    "nyc": "^11.0.2",
    "tap-xunit": "^1.7.0"
  },
  "nyc": {
    "sourceMap": false,
    "instrument": false
  },
  "babel": {
    "sourceMaps": "inline",
    "plugins": [
      "transform-es2015-parameters",
      "transform-es2015-template-literals",
      "transform-es2015-arrow-functions",
      "transform-es2015-spread"
    ],
    "env": {
      "test": {
        "plugins": [
          "istanbul"
        ]
      }
    }
  },
  "ava": {
    "tap": true,
    "require": [
      "babel-register"
    ],
    "babel": "inherit"
  }
}
