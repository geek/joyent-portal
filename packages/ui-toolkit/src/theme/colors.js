import React from 'react';
import styled from 'styled-components';
import { Strong } from 'normalized-styled-components';
import { Col, Row } from 'react-styled-flexboxgrid';
import remcalc from 'remcalc';
import titleCase from 'title-case';
import tinycolor from 'tinycolor2';
import { bottomShaddow } from '../boxes';
import P from '../text/p';
import theme from './';

const Box = styled.div`
  margin-bottom: ${remcalc(10)};
  box-shadow: ${bottomShaddow};
  border: solid ${remcalc(1)} ${props => props.border};
`;

const InnerBox = styled.div`
  padding: ${remcalc(18)};
  margin-top: ${remcalc(-7)};
  background: ${props => props.background};
  color: ${props => props.text};
`;
// Border: solid ${remcalc(1)} ${props => props.border};
// border-top-width: 0;

const Preview = styled.div`
  display: inline-block;
  background: ${props => props.hex};
  height: ${remcalc(100)};
  width: 100%;
`;

const Paragraph = P.extend`
  margin: 0;
`;

const baseColorNames = Object.keys(theme).filter(
  name => typeof theme[name] === 'string'
);

const mostReadable = hex =>
  tinycolor
    .mostReadable(hex, baseColorNames.map(name => theme[name]))
    .toHexString();

const borderColor = hex =>
  tinycolor
    .mostReadable(hex, [
      theme.text,
      theme.topologyBackground,
      theme.secondaryActive,
      theme.secondaryHover,
      theme.secondary
    ])
    .toHexString();

const Color = ({ name, hex }) =>
  <Box border={borderColor(hex)}>
    <Preview hex={hex} />
    <InnerBox background={hex} text={mostReadable(hex)}>
      <Paragraph>
        <Strong>Name</Strong>: <br />{titleCase(name)}
      </Paragraph>
      <br />
      <Paragraph>
        <Strong>Property</Strong>: <br /><code>{name}</code>
      </Paragraph>
      <br />
      <Paragraph>
        <Strong>Hex</Strong>: <br /><code>{hex.toUpperCase()}</code>
      </Paragraph>
    </InnerBox>
  </Box>;

export default () => {
  const colors = Object.keys(theme)
    .filter(name => typeof theme[name] === 'string')
    .sort((a, b) => {
      const _a = tinycolor(theme[a]).toHsl().h;
      const _b = tinycolor(theme[b]).toHsl().h;

      return _a >= _b ? -1 : 1;
    })
    .map(name =>
      <Col key={name} xs={4}>
        <Color name={name} hex={theme[name]} />
      </Col>
    );

  return (
    <Row>
      {colors}
    </Row>
  );
};
