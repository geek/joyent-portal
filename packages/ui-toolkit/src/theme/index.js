import remcalc from 'remcalc';

const flexboxgrid = {
  gridSize: 12, // rem
  gutterWidth: 1.25, // rem
  outerMargin: 1.875, // rem
  mediaQuery: 'only screen',
  container: {
    sm: 46, // rem
    md: 56, // rem
    lg: 59 // rem
  },
  breakpoints: {
    xs: 0, // em
    sm: 48, // em
    md: 64, // em
    lg: 75 // em
  }
};

const spacing = {
  0: remcalc(0),
  0.5: remcalc(3),
  1: remcalc(6),
  1.5: remcalc(9),
  2: remcalc(12),
  2.5: remcalc(15),
  3: remcalc(18),
  3.5: remcalc(21),
  4: remcalc(24),
  4.5: remcalc(27),
  5: remcalc(30),
  5.5: remcalc(33),
  6: remcalc(36),
  6.5: remcalc(39),
  7: remcalc(42),
  7.5: remcalc(45),
  8: remcalc(48),
  8.5: remcalc(51),
  9: remcalc(54),
  9.5: remcalc(57),
  10: remcalc(60),
  10.5: remcalc(63),
  11: remcalc(66),
  11.5: remcalc(69),
  12: remcalc(72),
  12.5: remcalc(75),
  13: remcalc(78),
  13.5: remcalc(81),
  '14px': remcalc(14)
};

/*
* There should be no string value to keys outside the `base` object.
* If a new colors needs to be used, check it doesn't already exist, or
* anything that is similar, and if it doesn't, add a key-value and reference from
* there. Lets try and keep different color variations down ot a minimum.
*
---
Color Object example
const color_name = {
  color_name: '#FFFFFF',
  color_name_style1: '#FFF000',
  color_name_style2: '#FFF333',
};
*/

const primary = {
  primary: 'rgb(59, 70, 204)',
  primaryHover: 'rgb(72, 83, 217)',
  primaryActive: 'rgb(45, 56, 132)'
};

// TOPOLOGY
const secondary = {
  secondary: 'rgb(70, 70, 70)',
  secondaryHover: 'rgb(53, 53, 53)',
  secondaryActive: 'rgb(45, 45, 45)'
};

const white = {
  white: 'rgb(255, 255, 255)',
  whiteHover: 'rgb(247, 247, 247)',
  whiteActive: 'rgb(230, 230, 230)'
};

const grey = {
  grey: 'rgb(216, 216, 216)',
  greyTransparent: 'rgba(73, 73, 73, 0.8)',
  greyLight: 'rgb(189, 189, 189)',
  greyDark: 'rgb(151, 151, 151)',
  greyDarker: 'rgb(052, 052, 052)'
};

const green = {
  green: 'rgb(0, 152, 88)',
  greenDark: 'rgb(0, 129, 56)'
};

const orange = {
  orange: 'rgb(227, 130, 0)',
  orangeDark: 'rgb(203, 116, 0)'
};

const red = {
  red: 'rgb(210, 67, 58)',
  redDark: 'rgb(205, 37, 27)'
};

/** ********************************** BASE *********************************** */

export const base = {
  ...primary,
  ...secondary,
  ...white,
  ...red,
  ...orange,
  ...green,
  ...grey,
  text: 'rgba(73, 73, 73, 1)',
  textDisabled: 'rgba(73, 73, 73, 0.5)',
  placeholder: 'rgb(99, 99, 99)',
  disabled: 'rgb(250, 250, 250)', // used
  background: 'rgb(250, 250, 250)' // used
};

/** ********************************** HEADER ********************************** */

const brandBackground = 'rgb(30, 49, 59)';

/** ********************************** FONTS ********************************** */

export const font = {
  semibold: base.secondary,
  regular: base.text,
  abbrBorderColor: base.secondary,
  textMuted: base.secondary,
  family: '"Libre Franklin"',
  families:
    '"Libre Franklin", -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, Helvetica, sans-serif',
  href: () =>
    `${document.location.protocol}//${
      document.location.host
    }/fonts/css?family=Libre+Franklin:400,500,600,700`,
  weight: {
    bold: 700,
    semibold: 600,
    medium: 500,
    normal: 400
  }
};

export const monoFont = {
  semibold: base.secondary,
  regular: base.text,
  abbrBorderColor: base.secondary,
  textMuted: base.secondary,
  family: '"Roboto Mono"',
  families: '"Roboto Mono", monospace',
  href: () =>
    `${document.location.protocol}//${
      document.location.host
    }/fonts/css?family=Roboto+Mono:700,400`,
  weight: {
    bold: 700,
    normal: 400
  }
};

/** ******************************** INACTIVE ********************************* */

export const inactive = {
  default: base.disabled,
  border: base.grey,
  text: base.grey
};

/** ***************************** NOTIFICATIONS ******************************* */

export const notifications = {
  alert: base.red,
  confirmation: base.green,
  warning: base.orange
};

/** ********************************* FORMS *********************************** */

export const inputError = base.red;
export const inputWarning = base.orange;

/** ******************************** METRICS ********************************* */

/* export const miniBackground = '#F3F4F9';
export const separator = '#D9DEF3'; */

/** ******************************** TOPOLOGY ********************************* */

export const topologyBackground = base.secondaryActive;

export const transition = 'all 200ms ease-out';

export const borderRadius = remcalc(4);

export const shadows = {
  bottomShadow: `0 ${remcalc(2)} 0 0 rgba(0, 0, 0, 0.05)`,
  bottomShadowDarker: `0 ${remcalc(2)} 0 0 rgba(0, 0, 0, 0.1)`
};

export default {
  ...base,
  flexboxgrid,
  spacing,
  font,
  monoFont,
  inactive,
  notifications,
  inputError,
  inputWarning,
  topologyBackground,
  brandBackground,
  transition,
  borderRadius,
  shadows,
  transparent: 'transparent'
};
