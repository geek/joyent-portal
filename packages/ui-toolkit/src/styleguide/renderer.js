import React from 'react';
import styled, { ThemeProvider } from 'styled-components';
import remcalc from 'remcalc';
import is from 'styled-is';

import { Grid, Row, Col } from '../grid';
import theme from '../theme';

const Main = styled(Row)`
  padding-top: ${remcalc(40)};
`;

const Sticky = styled.div`
  position: sticky;
  top: 0;
  z-index: 9;

  ${is('sidebar')`
    top: ${remcalc(100)};
    max-height: 85vh;
    overflow: auto;
  `};
`;

const fullTheme = {
  ...theme,
  spacing: {
    0.5: remcalc(4),
    0: remcalc(0),
    1: remcalc(6),
    2: remcalc(12),
    3: remcalc(18),
    4: remcalc(24),
    5: remcalc(30),
    6: remcalc(36),
    7: remcalc(42),
    8: remcalc(48),
    9: remcalc(54),
    10: remcalc(60)
  }
};

const StyleGuideRenderer = ({
  title,
  homepageUrl,
  children,
  toc,
  hasSidebar
}) => (
  <ThemeProvider theme={fullTheme}>
    <Grid>
      <Main>
        {hasSidebar && (
          <Col xs={3}>
            <Sticky sidebar>{toc}</Sticky>
          </Col>
        )}
        <Col xs={hasSidebar ? 9 : 12} lg={hasSidebar ? 8 : 12}>
          {children}
        </Col>
      </Main>
    </Grid>
  </ThemeProvider>
);

export default StyleGuideRenderer;
