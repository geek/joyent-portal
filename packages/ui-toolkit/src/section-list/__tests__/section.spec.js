import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import remcalc from 'remcalc';

import SectionList, { Item, Anchor } from '../';
import { Theme } from '../../mocks';
import theme from '../../theme';

describe('SectionList', () => {
  test('SectionList', () => {
    const tree = renderer
      .create(
        <Theme>
          <SectionList>
            <Item>
              <Anchor active>Overview</Anchor>
            </Item>
            <Item>
              <Anchor>Services</Anchor>
            </Item>
            <Item>
              <Anchor>Instances</Anchor>
            </Item>
          </SectionList>
        </Theme>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Item', () => {
    const tree = renderer
      .create(
        <Theme>
          <Item>
            <Anchor>Instances</Anchor>
          </Item>
        </Theme>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('font-size', remcalc(15));
  });

  test('Anchor', () => {
    const tree = renderer
      .create(
        <Theme>
          <Anchor>Instances</Anchor>
        </Theme>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('color', theme.secondary.replace(/ /g, ''));
  });

  test('Anchor', () => {
    const tree = renderer
      .create(
        <Theme>
          <Anchor active>Instances</Anchor>
        </Theme>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('color', theme.secondary.replace(/ /g, ''));
  });
});
