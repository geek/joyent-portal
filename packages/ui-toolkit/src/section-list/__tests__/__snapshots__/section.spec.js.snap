// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionList Anchor 1`] = `
.c2 {
  background-color: transparent;
  -webkit-text-decoration-skip: objects;
  text-decoration-skip: objects;
  font-family: "Libre Franklin",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica,sans-serif;
}

.c1 {
  color: rgb(59,70,204);
  -webkit-text-fill-color: currentcolor;
}

.c1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  color: rgb(70,70,70);
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  padding: 0 0.375rem;
  padding-bottom: 1rem;
  position: relative;
}

.c0.active {
  font-weight: 600;
}

.c0.active:after {
  width: 100%;
  height: 0.0625rem;
  background: rgb(70,70,70);
  content: '';
  position: absolute;
  bottom: 0;
  left: 0;
  z-index: 2;
}

<a
  className="c0 c1 c2"
>
  Instances
</a>
`;

exports[`SectionList Anchor 2`] = `
.c2 {
  background-color: transparent;
  -webkit-text-decoration-skip: objects;
  text-decoration-skip: objects;
  font-family: "Libre Franklin",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica,sans-serif;
}

.c1 {
  color: rgb(59,70,204);
  -webkit-text-fill-color: currentcolor;
}

.c1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  color: rgb(70,70,70);
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  padding: 0 0.375rem;
  padding-bottom: 1rem;
  position: relative;
  font-weight: 600;
}

.c0.active {
  font-weight: 600;
}

.c0.active:after {
  width: 100%;
  height: 0.0625rem;
  background: rgb(70,70,70);
  content: '';
  position: absolute;
  bottom: 0;
  left: 0;
  z-index: 2;
}

.c0:after {
  width: 100%;
  height: 0.0625rem;
  background: rgb(70,70,70);
  content: '';
  position: absolute;
  bottom: 0;
  left: 0;
  z-index: 2;
}

<a
  className="c0 c1 c2"
>
  Instances
</a>
`;

exports[`SectionList Item 1`] = `
.c3 {
  background-color: transparent;
  -webkit-text-decoration-skip: objects;
  text-decoration-skip: objects;
  font-family: "Libre Franklin",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica,sans-serif;
}

.c2 {
  color: rgb(59,70,204);
  -webkit-text-fill-color: currentcolor;
}

.c2:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  font-size: 0.9375rem;
  line-height: 1.6;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0:not(:last-child) {
  margin-right: 1.875rem;
}

.c1 {
  color: rgb(70,70,70);
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  padding: 0 0.375rem;
  padding-bottom: 1rem;
  position: relative;
}

.c1.active {
  font-weight: 600;
}

.c1.active:after {
  width: 100%;
  height: 0.0625rem;
  background: rgb(70,70,70);
  content: '';
  position: absolute;
  bottom: 0;
  left: 0;
  z-index: 2;
}

<li
  className="c0"
>
  <a
    className="c1 c2 c3"
  >
    Instances
  </a>
</li>
`;

exports[`SectionList SectionList 1`] = `
.c4 {
  background-color: transparent;
  -webkit-text-decoration-skip: objects;
  text-decoration-skip: objects;
  font-family: "Libre Franklin",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica,sans-serif;
}

.c3 {
  color: rgb(59,70,204);
  -webkit-text-fill-color: currentcolor;
}

.c3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  font-size: 0.9375rem;
  line-height: 1.6;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c1:not(:last-child) {
  margin-right: 1.875rem;
}

.c2 {
  color: rgb(70,70,70);
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  padding: 0 0.375rem;
  padding-bottom: 1rem;
  position: relative;
  font-weight: 600;
}

.c2.active {
  font-weight: 600;
}

.c2.active:after {
  width: 100%;
  height: 0.0625rem;
  background: rgb(70,70,70);
  content: '';
  position: absolute;
  bottom: 0;
  left: 0;
  z-index: 2;
}

.c2:after {
  width: 100%;
  height: 0.0625rem;
  background: rgb(70,70,70);
  content: '';
  position: absolute;
  bottom: 0;
  left: 0;
  z-index: 2;
}

.c5 {
  color: rgb(70,70,70);
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  padding: 0 0.375rem;
  padding-bottom: 1rem;
  position: relative;
}

.c5.active {
  font-weight: 600;
}

.c5.active:after {
  width: 100%;
  height: 0.0625rem;
  background: rgb(70,70,70);
  content: '';
  position: absolute;
  bottom: 0;
  left: 0;
  z-index: 2;
}

.c0 {
  background: rgb(250,250,250);
  list-style-type: none;
  padding: 0.8125rem 0.75rem;
  margin: 1.125rem 0 0 0;
  min-width: 100%;
  max-height: 50px;
  overflow-x: scroll;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.c0:after {
  width: 100%;
  height: 0.0625rem;
  background: rgb(216,216,216);
  content: '';
  position: absolute;
  bottom: 0;
  left: 0;
  z-index: 1;
}

<ul
  className="c0"
>
  <li
    className="c1"
  >
    <a
      className="c2 c3 c4"
    >
      Overview
    </a>
  </li>
  <li
    className="c1"
  >
    <a
      className="c5 c3 c4"
    >
      Services
    </a>
  </li>
  <li
    className="c1"
  >
    <a
      className="c5 c3 c4"
    >
      Instances
    </a>
  </li>
</ul>
`;
